HelperFunctions/uitools/uifitgeotrans.m

```diff
@@ -0,0 +1,330 @@
+n = min([nFixed,nMoving]);
+fitgeotrans(movingPoints(1:n,:),fixedPoints(1:n,:),transformationType,varargin{:});
+
+lastCloseRequestFcn = 'closereq'; % default
+lastWindowKeyPressFcn = '';
+lastButtonDownFcn = '';
+lastTitle = '';
+if ~isempty(p.Parent)
+    ax = p.Parent;
+    f = Base.getParentFigure(ax);
+    lastCloseRequestFcn = f.CloseRequestFcn;
+    lastWindowKeyPressFcn = f.WindowKeyPressFcn;
+    lastButtonDownFcn = ax.ButtonDownFcn;
+    
+    lastTitle = ax.Title.String;
+else
+    f = figure();
+    ax = axes('parent',f);
+end
+f.CloseRequestFcn = @confirm_close;
+f.WindowKeyPressFcn = @keyPressed;
+ax.ButtonDownFcn = @clicked;
+hold(ax,'on');
+try
+    for i = 1:nMoving
+        points{1}(i) = line(ax, movingPoints(i,1), movingPoints(i,2),...
+            'marker','x','linestyle','none','color',cs(1,:),'tag','point',...
+            'UserData',struct('type',1,'ind',i,'origPos',movingPoints(i,:)),...
+            movingLineParams{:});
+    end
+    for i = 1:nFixed
+        points{2}(i) = line(ax, fixedPoints(i,1), fixedPoints(i,2),...
+            'marker','o','linestyle','none','color',cs(2,:),'tag','point',...
+            'UserData',struct('type',2,'ind',i),...
+            fixedLineParams{:});
+    end
+catch err
+     if isempty(p.Parent) % We generated, so clean up
+         delete(f);
+     else
+        delete(points{1}); delete(points{2});
+        f.CloseRequestFcn = lastCloseRequestFcn;
+     end
+     rethrow(err);
+end
+leg = legend([points{1}(1),points{2}(1)],point_types,...
+    'Orientation','horizontal','location','northoutside');
+
+% Add context menu
+cm = uicontextmenu(f);
+% UserData is effectively zero-indexed into "points" cell array and menu
+% list. Identical to "state"
+menu(1) = uimenu(cm,'label','Select moving point','callback',@selectPointType,'UserData',0);
+menu(2) = uimenu(cm,'label','Select fixed point','callback',@selectPointType,'UserData',1);
+uimenu(cm,'label','Finished','callback',@finished,'separator','on');
+f.UIContextMenu = cm;
+ax.UIContextMenu = cm;
+set(points{1},'UIContextMenu',cm,'ButtonDownFcn',@clicked);
+set(points{2},'UIContextMenu',cm,'ButtonDownFcn',@clicked);
+% Context menu to remove selected ones
+cm_remove = uicontextmenu(f);
+uimenu(cm_remove,'label','Remove from paired points','callback',@removePointPair);
+
+% Set up default behavior
+selectPointType(menu(state+1)); % "Activates" moving points
+title('Select active moving point.');
+uiwait(f);
+if ~isvalid(f)
+    return;
+end
+aborted = false;
+if p.PreserveFig
+    remove_interactivity();
+else
+    delete(f);
+end
+
+    % Callbacks/helpers
+    function finished(~,~)
+        if ~altered
+            resp = questdlg('No changes have been made yet. Return identity tform?',...
+                            'uifitgeotrans: No Changes Detected','Yes','Cancel','Cancel');
+            if strcmp(resp,'Cancel')
+                return
+            end
+        end
+        uiresume(f);
+    end
+    function clicked(~,eventdata)
```

Comment:
With this configuration, the order in which you choose the active points swaps each round (i.e. if you start choosing a fixed point and then a moving point, in the next round you choose a moving point then a fixed point). Was this on purpose?

Reply:
Nope - just a byproduct of the logic. Main purpose was to automatically switch when necessary and stay when ready to link points. Previously you could only link based on active set of points.  That has since been changed and generalized to axes, so we can change to swap each time.